lib/cfn-nag/custom_rules/IamRoleAdministratorAccessPolicyRule.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+require 'cfn-nag/violation'
+require_relative 'base'
+
+class IamRoleAdministratorAccessPolicyRule < BaseRule
+  def rule_text
+    'IAM role should not have AdministratorAccess policy'
+  end
+
+  def rule_type
+    Violation::WARNING
+  end
+
+  def rule_id
+    'W25'
+  end
+
+  def audit_impl(cfn_model)
+    violating_roles = cfn_model.resources_by_type('AWS::IAM::Role').select do |role|
+      violating_policies = role.managedPolicyArns.select do |policy|
```

Comment:
this is fine, but you could just do a "find" instead - quit after the first one since it only takes one to violate